src/components/sidebar/CallTreeSidebar.js

```diff
@@ -40,26 +45,76 @@ function SidebarDetail({ label, children }: SidebarDetailProps) {
   );
 }
 
+type ImplementationBreakdownProps = {|
+  +breakdown: BreakdownByImplementation,
+|};
+
+// This component is responsible for displaying the breakdown data specific to
+// the JavaScript engine and native code implementation.
+class ImplementationBreakdown extends React.PureComponent<
+  ImplementationBreakdownProps
+> {
+  _orderedImplementations: $ReadOnlyArray<StackImplementation> = [
+    'native',
+    'interpreter',
+    'baseline',
+    'ion',
+    'unknown',
+  ];
+
+  _labelizeImplementation(implementation: StackImplementation): string {
+    switch (implementation) {
+      case 'ion':
+      case 'baseline':
+        return `JavaScript JIT (${implementation})`;
+      case 'interpreter':
+        return 'JavaScript interpreter';
+      case 'native':
+        return 'Native code';
+      case 'unknown':
+        return implementation;
+      default:
+        throw assertExhaustiveCheck(implementation);
+    }
+  }
+
+  render() {
+    const { breakdown } = this.props;
+    const data = [];
+
+    for (const implementation of this._orderedImplementations) {
+      const value = breakdown[implementation];
+      if (!value && implementation === 'unknown') {
+        continue;
+      }
+
+      data.push({
+        group: this._labelizeImplementation(implementation),
+        value: value || 0,
+      });
```

Comment:
I'm not so sure about this: should we display the implementations even when their value is 0 ?

Reply:
ok, I'll remove the related code then.